pwndbg/emu/emulator.py

```diff
@@ -574,16 +577,19 @@ def get_uc_mode(self):
         if arch == "armcm":
             mode |= (
                 (U.UC_MODE_MCLASS | U.UC_MODE_THUMB)
-                if (pwndbg.gdblib.regs.xpsr & (1 << 24))
+                if (pwndbg.aglib.regs.xpsr & (1 << 24))
                 else U.UC_MODE_MCLASS
             )
 
         elif arch in ("arm", "aarch64"):
-            mode |= U.UC_MODE_THUMB if (pwndbg.gdblib.regs.cpsr & (1 << 5)) else U.UC_MODE_ARM
+            mode |= U.UC_MODE_THUMB if (pwndbg.aglib.regs.cpsr & (1 << 5)) else U.UC_MODE_ARM
 
-        elif arch == "mips" and "isa32r6" in gdb.newest_frame().architecture().name():
+        elif (
+            arch == "mips"
+            and pwndbg.dbg.is_gdblib_available()
+            and "isa32r6" in gdb.newest_frame().architecture().name()
```

Comment:
Why gdb check here? does lldb not support mips?

Reply:
We'll address the `gdb.newest_frame().architecture().name()` calls later as they appear in many places. For now, this is a 'best effort' implementation 🤔

Reply:
We just here don't support `MIPS32R6` arch for lldb